tests/tools/traffic-replay/RandomReplay.py

```diff
@@ -0,0 +1,167 @@
+import gzip
+bSTOP = False
+def createDummyBodywithLength(numberOfbytes):
+    if numberOfbytes<=0:
+        return None
+    body= 'a'
+    while numberOfbytes!=1:
+        body += 'b'
+        numberOfbytes -= 1
+    return body
+    
+def handleResponse(response,*args, **kwargs):
+    print(response.status_code)
+    #resp=args[0]
+    #expected_output_split = resp.getHeaders().split('\r\n')[ 0].split(' ', 2)
+    #expected_output = (int(expected_output_split[1]), str( expected_output_split[2]))
+    #r = result.Result(session_filename, expected_output[0], response.status_code)
+    #print(r.getResultString(colorize=True))
+# make sure len of the message body is greater than length
+def gen():
+    yield 'pforpersia,champaignurbana'.encode('utf-8')
+    yield 'there'.encode('utf-8')
+
+def txn_replay(session_filename, txn, proxy, result_queue, request_session):
+    """ Replays a single transaction
+    :param request_session: has to be a valid requests session"""
+    req = txn.getRequest()
+    resp = txn.getResponse()
+
+    # Construct HTTP request & fire it off
+    txn_req_headers = req.getHeaders()
+    txn_req_headers_dict = extractHeader.header_to_dict(txn_req_headers)
+    txn_req_headers_dict['Content-MD5'] = txn._uuid  # used as unique identifier
+    if 'body' in txn_req_headers_dict:
+        del txn_req_headers_dict['body']
+    
+    #print("Replaying session")
+    try:
+        #response = request_session.request(extractHeader.extract_txn_req_method(txn_req_headers),
+        #                            'http://' + extractHeader.extract_host(txn_req_headers) + extractHeader.extract_GET_path(txn_req_headers),
+        #                            headers=txn_req_headers_dict,stream=False) # making stream=False raises contentdecoding exception? kill me
+        method = extractHeader.extract_txn_req_method(txn_req_headers)
+        response = None
+        body=None
+        content=None
+        if 'Transfer-Encoding' in txn_req_headers_dict:
+            # deleting the host key, since the STUPID post/get functions are going to add host field anyway, so there will be multiple host fields in the header
+            # This confuses the ATS and it returns 400 "Invalid HTTP request". I don't believe this
+            # BUT, this is not a problem if the data is not chunked encoded.. Strange, huh?
+            del txn_req_headers_dict['Host']
+            if 'Content-Length' in txn_req_headers_dict:
+                #print("ewww !")
+                del txn_req_headers_dict['Content-Length']
+                body = gen()
+        if 'Content-Length' in txn_req_headers_dict:
+            nBytes=int(txn_req_headers_dict['Content-Length'])
+            body = createDummyBodywithLength(nBytes)
+        #print("request session is",id(request_session))
+        if method == 'GET':     
+            response = request_session.get('http://' + extractHeader.extract_host(txn_req_headers) + extractHeader.extract_GET_path(txn_req_headers),
```

Comment:
OK for now but it will definitely be requested that the replay can replay both "normal" requests and "proxy style" requests (the latter meaning the full URL is in the `GET`).